deploy_rln_contract.sh

```diff
@@ -0,0 +1,42 @@
+
```

Comment:
The forge command is available, but it depends on whether there are other dependencies as well, such as hardhat. Then it gets a bit more difficult.
We could add the waku-rln-contract repo as a submodule and extract the bytecode from the json files in the /deployments/sepolia directory?

Reply:
Do we need to add it as a submodule? Or perhaps in the script we can do "git clone xxx" + "git checkout yyy"?

I'm open about the "how" but would be nice to be able to set a specific waku-rln contract in an easy way with a simpler script.

Reply:
> Do we need to add it as a submodule? Or perhaps in the script we can do "git clone xxx" + "git checkout yyy"?
> 

this should work, after installing the dependencies, and running `yarn deploy:localhost`

Reply:
if we can use the commit hash in the curl, it should be better, in case it changes over time

Reply:
Yes, this is possible, 'main' in the above URL can be replaced with the commit hash

Reply:
I implemented retrieval of the data from github using curl in [ade94d3](https://github.com/waku-org/waku-simulator/pull/30/commits/ade94d328b57a2227ec0d28e6497827060607727).
Should I make the branch/commit hash in the URL path an environment variable?

Reply:
@rymnc is it possible to simplify this script even further? I mean git cloning waku-rln-contract and then running `yarn deploy:localhost`?

Don't fully understand the need to rewrite a script like this one, when we already have one in waku rln repo. Also imagine that we change rln contract deployment (eg new contract). Then we would need to update it here as well, etc.

Unsure if this can be done within anvil, + needed dependencies, but using another container for this is ok. Eg: new docker service running a simple linux image that git clones and yarn deploys, and exits 0.

Not saying it should be we way but IMHO its important to simplify this and reuse as much as we can.

Reply:
I completely agree, that was the approach i suggested initially.

Reply:
What about a nodejs docker image which runs a script that:
- installs foundry
- clones the git repo
- builds the code
- executes 'yarn deploy:localhost'

Reply:
that sounds good. does "yarn install" inside the rln contract repo install required dependencies?

Reply:
'yarn install' will install the hardhat dependencies.
foundry is needed for the 'forge install' which retrieves required submodules.

I have implemented this but the new container which clones and builds the waku-rln-contract code doesn't seem to have access to the anvil container using http://localhost:8545, but running a RPC call using curl it does work using the name of the anvil container (foundry).

This works:
```curl http://foundry:8545 -X POST -H "Content-Type: application/json" --data '{"method":"eth_chainId","params":[],"id":1,"jsonrpc":"2.0"}'
```
This doesn't:
```curl http://localhost:8545 -X POST -H "Content-Type: application/json" --data '{"method":"eth_chainId","params":[],"id":1,"jsonrpc":"2.0"}'
```

To solve this I could merge the functionality of the two containers into one container.
Otherwise it would be nice if the RPC URL could be provided as an argument for the yarn deploy, but then we would need to modify the deployment script in the waku-rln-contract repo.

Reply:
If we add the rpc url to the hardhat.config.ts

```
const getNetworkConfig = (): NetworksUserConfig | undefined => {
  if (SEPOLIA_URL && PRIVATE_KEY) {
    return {
      sepolia: {
        url: SEPOLIA_URL,
        accounts: [PRIVATE_KEY],
        forking: {
          url: SEPOLIA_URL,
        },
        verify: {
          etherscan: {
            apiKey: ETHERSCAN_API_KEY,
            apiUrl: "https://api-sepolia.etherscan.io",
          },
        },
      },
      localhost_integration: {
        url: "http://localhost:8545",
      },
      anvil: { url: ANVIL_RPC_URL },
    };
  }
  return undefined;
};
```
it will work with `yarn deploy anvil`

Reply:
Will this require many additional changes in the waku-rln-contract repo?

Reply:
imho "yarn install" should allow to pass an env RPC_PROVIDER to override the localhost?

Reply:
The contract deployment using waku-rln-contract repo's yarn deploy command from inside a separate docker container implemented in [8eb3908](https://github.com/waku-org/waku-simulator/pull/30/commits/8eb3908ed79dfbc914293ce885171504d1a08b39)

Reply:
Does that work on your machine? I'm getting this.

```
docker-compose logs contract-repo-deployer
waku-simulator-contract-repo-deployer-1  | /usr/local/bin/docker-entrypoint.sh: 11: exec: /opt/deploy_rln_contract.sh: Permission denied
```

Reply:
It does work on my machine.
Could you try adding `entrypoint: sh` in the docker-compose file?

Reply:
seems to work in my machine now thanks!